src/marqo/core/structured_vespa_index/structured_vespa_index.py

```diff
@@ -500,6 +508,52 @@ def _verify_marqo_field_type(self, field_name: str, value: Any):
                                        f'{marqo_type.value}. Expected a value of type {python_type}, but found '
                                        f'{type(value)}')
 
+    def _verify_numerical_field_value(self, value: Union[float, int], index_field: Field):
+        if index_field.type in (FieldType.Float, FieldType.ArrayFloat):
+            self._verify_float_field_range(value)
+        elif index_field.type in (FieldType.Int, FieldType.ArrayInt):
+            self._verify_int_field_range(value)
+        elif index_field.type in (FieldType.Long, FieldType.ArrayLong):
+            self._verify_long_field_range(value)
+        elif index_field.type in (FieldType.Double, FieldType.ArrayDouble):
+            pass
+        else:
+            raise InternalError(f'Invalid field type {index_field.type} for field {index_field.name} called by'
+                                f'_verify_numerical_field_value. Expected one of {FieldType.Float}, {FieldType.Int}, '
+                                f'{FieldType.Long}, {FieldType.Double}.')
+
+    def _verify_float_field_range(self, value: float):
+        MAX_FLOAT = 3.4028235e38
+        MIN_FLOAT = -3.4028235e38
+        SMALLEST_POS_FLOAT = 1.4e-45
+        LARGEST_NEG_FLOAT = -1.4e-45
+        if (value > MAX_FLOAT) or (0 < value < SMALLEST_POS_FLOAT) or (value < MIN_FLOAT) or \
```

Comment:
What happens if `< SMALLEST_POS_FLOAT` or `> LARGEST_NEG_FLOAT`? If truncation/rounding, I think that is acceptable and we shouldn't error out for that